# Regresión Lineal Múltiple - CincinnatiPaintCompany

1. Determinal las variables independientes
2. Determinar la variable dependiente
3. Obtener el valor de intesección
4. Responder las preguntas

In [13]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [14]:
#importar datos
df = pd.read_csv("CincinnatiPaintCompany.csv")
df.head()

Commissions  Calls  Driven
0           22    139    2371
1           13    132    2226
2           33    144    2731
3           38    142    3351
4           23    142    2289

### 1. Determinar las variables independientes
x1 = calls, x2 = driven

In [15]:
X = df[['Calls','Driven']]

### 2. Determinar las variables dependientes Y
y = commissions

In [16]:
Y = df[['Commissions']]

In [17]:
#Ajustar el model de regresión multiple
model = LinearRegression()
model.fit(X, Y) #Entrena el modelo
y_pred = model.predict(X)
#y_pred

#Métricas estadísticas
mse = mean_squared_error(Y, y_pred)
r2 = r2_score(Y, y_pred)

### 3. Valor de intersección

In [18]:
model.intercept_

array([-101.31444968])

In [19]:
#coeficientes
print('Coeficientes: ', model.coef_)
print("Error cuadrático medio ", mse)
print("Varianza: ", r2)

Coeficientes:  [[0.63233256 0.01568925]]
Error cuadrático medio  16.313412727936065
Varianza:  0.8352686374291524


### 4. Responder las preguntas

**1. ¿Ganan más comisiones los agentes que recorren más millas y hacen más llamadas?**   
    Si pero casi exclusivamente por las llamadas, ya que las millas recorridas influyen muy poco


**2. ¿Hay una interacción significativa entre el numero de llamadas y millas recorridas?**
    El 0.63 representa a las llamadas hechas, significa que influye positivamente y de manera moderada por su alto valor.   
    El 0.01 nos indica que, aunque positiva hay muy poca influencia directa por este valor.
